### PR TITLE
feat: access-level middleware

### PR DESCRIPTION
**Pull request by AasmundN**
# Changes
- Extend route meta data types.
- Add accessLevel getter to auth store to check the users accessLevel.
- Update route middleware to check accessLevel.

### CODE REVIEW

File: stores/authStore.ts [hunk 1]
```diff
@@ -12,5 +12,7 @@ export const useAuthStore = defineStore('auth', {
   },
   getters: {
     isLoggedIn: (state) => !!state.user,
+    hasAccessLevel: (state) => (accessLevel?: string) =>
+      accessLevel ? state.user?.accessLevel?.includes(accessLevel) : true,
```
Comment (ipeglin) [inline stores/authStore.ts:16]: LGTM. Maybe go through this syntax with the other group members at some point? Or leave a comment. The `?` syntax might look a bit scary coming from Python :)

File: models/Meta.d.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+// extend route meta data types
+
+import 'vue-router'
+
+// To ensure it is treated as a module, add at least one `export` statement
```
Comment (ipeglin) [inline models/Meta.d.ts:5]: Weird. Even when explicitly declaring a module underneath?

File: middleware/auth.global.ts [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
-  // route is protected
+  // check if route is protected
   if (!to.meta.protected) return
 
-  // when navigating client side the auth state is
-  // collected from the auth store
-  const isLoggedIn = useAuthStore().isLoggedIn
+  // get auth state from store
+  const auth = useAuthStore()
+  const isLoggedIn = auth.isLoggedIn
```
Comment (ipeglin) [inline middleware/auth.global.ts:7]: Clean code. I like

**Review by ipeglin — APPROVED** [review]
LGTM

File: middleware/auth.global.ts [hunk 4]
```diff
@@ -1,13 +1,15 @@
...
 
-  // user is authenticated
-  if (isLoggedIn) return
+  const accessLevel = to.meta.accessLevel
+
+  // user is authenticated and has the required access level
+  if (isLoggedIn && auth.hasAccessLevel(accessLevel)) return
 
   // user is not authenticated
   return navigateTo('/')
```
[the remark this answers is not in the document]
Comment (AasmundN) [reply middleware/auth.global.ts:15]: Yes. Will change it

File: models/Meta.d.ts (hunk 2 above)
Comment (AasmundN) [reply models/Meta.d.ts:5]: Apparently. Copied from the docs.

**Pull request merged**